html/pfappserver/lib/pfappserver/Form/Config/PKI_Provider.pm

```diff
@@ -0,0 +1,104 @@
+package pfappserver::Form::Config::PKI_Provider;
+
+=head1 NAME
+
+pfappserver::Form::Config::PKI_Provider
+
+=head1 DESCRIPTION
+
+=cut
+
+use strict;
+use warnings;
+use HTTP::Status qw(:constants is_error is_success);
+use HTML::FormHandler::Moose;
+extends 'pfappserver::Base::Form';
+with 'pfappserver::Base::Form::Role::Help';
+
+use pf::log;
+
+use pf::factory::pki_provider;
+
+## Definition
+has_field 'id' =>
+  (
+   type => 'Text',
+   label => 'PKI Provider Name',
+   required => 1,
+   messages => { required => 'Please specify the name of the PKI provider' },
+  );
+
+has_field 'type' =>
+  (
+   type => 'Select',
+   required => 1,
+   messages => { required => 'Parameter 1 is required.' },
+   options => [map { { label => $_, value => $_ } } qw(inverse) ],
+  );
+
+has_field 'uri' =>
```

Comment:
This needs a help message. I'd be clueless if I was looking at this form right now